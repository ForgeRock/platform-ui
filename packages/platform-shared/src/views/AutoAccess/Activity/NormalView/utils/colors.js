/**
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

export const colorsHex = ['#2ed47a', '#ffb946', '#109cf1', '#f7685b', '#61e2e2'];
export const colorsReadable = ['green', 'yellow', 'blue', 'red', 'cyan'];
