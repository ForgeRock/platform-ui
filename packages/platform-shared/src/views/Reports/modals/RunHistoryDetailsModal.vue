<!-- Copyright (c) 2023 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <BModal
    v-model="showModal"
    data-testid="run-history-details-modal"
    ok-only
    ok-variant="outline-primary"
    size="lg"
    :ok-title="$t('common.close')"
    :static="isTesting"
    :title="$t('reports.tabs.runHistory.table.runDetails')">
    <BContainer class="p-0 text-dark">
      <BRow no-gutters>
        <BCol md="12">
          <BListGroup horizontal>
            <BListGroupItem class="border-0 pt-0 pl-0 pr-5">
              <small class="text-muted">
                {{ $t('common.started') }}
              </small>
              <div data-testid="fr-run-history-summary-modal-date">
                {{ dayjs(tableItem.date).format('MM/DD/YYYY h:mm A') }}
              </div>
            </BListGroupItem>
            <BListGroupItem class="border-0 pt-0">
              <div>
                <small class="text-muted">
                  {{ $t('common.status') }}
                </small>
              </div>
              <FrRunReportBadges :report-status="tableItem.reportStatus" />
            </BListGroupItem>
          </BListGroup>
          <FrReportDetails :parameters="props.parameters" />
        </BCol>
      </BRow>
    </BContainer>
  </BModal>
</template>

<script setup>
/**
 * @description
 * Matches the incoming parameters with the list of all fields
 * and constructs a dataset for displaying the original run
 * details that the user input to generate a report.
 */
import { computed } from 'vue';
import {
  BCol,
  BContainer,
  BListGroup,
  BListGroupItem,
  BModal,
  BRow,
} from 'bootstrap-vue';
import dayjs from 'dayjs';
import FrRunReportBadges from '../RunReportBadges';
import FrReportDetails from '../ReportDetails';

const props = defineProps({
  isTesting: {
    type: Boolean,
    default: false,
  },
  parameters: {
    type: Object,
    default: () => ({}),
  },
  show: {
    type: Boolean,
    default: false,
  },
  tableItem: {
    type: Object,
    default: () => ({}),
  },
});

const emit = defineEmits(['modal-toggle']);

const showModal = computed({
  get() {
    return props.show;
  },
  set(newValue) {
    emit('modal-toggle', newValue);
  },
});
</script>
