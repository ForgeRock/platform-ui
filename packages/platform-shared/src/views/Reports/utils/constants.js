/**
 * Copyright (c) 2025 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

/**
 * Report viewer group name
 */
export const REPORT_VIEWER_GROUP = 'report_viewer';

/**
 * Used for mapping field type names that come from the API,
 * to the field type names that the <FrField> component consumes.
 */
export const FIELD_TYPE_MAP = {
  array: 'multiselect',
  boolean: 'boolean',
  date: 'date',
  float: 'number',
  integer: 'number',
  string: 'string',
};
