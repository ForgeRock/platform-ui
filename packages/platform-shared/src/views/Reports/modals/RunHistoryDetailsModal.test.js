/**
 * Copyright (c) 2023-2025 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { nextTick } from 'vue';
import { mount } from '@vue/test-utils';
import { findByTestId, findByRole } from '@forgerock/platform-shared/src/utils/testHelpers';
import dayjs from 'dayjs';
import i18n from '@/i18n';
import RunHistoryDetailsModal from './RunHistoryDetailsModal';
import useRunReport from '../composables/RunReport';

const { _PARAMETERS_CONTROLLER } = useRunReport();
const allFieldKeys = Object.keys(_PARAMETERS_CONTROLLER);
const allFieldKeysObject = allFieldKeys.map((key) => {
  const hasFetch = _PARAMETERS_CONTROLLER[key]?.config?.fetch;
  return { [key]: hasFetch ? [key] : key };
});
const parameters = {};
allFieldKeysObject.forEach((obj) => {
  const key = Object.keys(obj);
  const value = Object.values(obj)[0];
  parameters[key] = value;
});

describe('Run History Details modal component', () => {
  function setup(props) {
    return mount(RunHistoryDetailsModal, {
      global: {
        plugins: [i18n],
      },
      props: {
        isTesting: true,
        parameters,
        show: true,
        tableItem: {
          date: '10/15/2023',
          reportStatus: 'complete',
        },
        ...props,
      },
    });
  }

  let wrapper;

  describe('@component', () => {
    beforeEach(async () => {
      wrapper = setup();
      await nextTick();
    });

    it('emits "modal-toggle" when the "showModal" computed property is updated', async () => {
      const modalElementShowing = findByRole(wrapper, 'dialog');
      expect(modalElementShowing.isVisible()).toBe(true);

      await wrapper.setProps({ show: false });
      expect(wrapper.vm.showModal).toEqual(false);

      const modalElementHidden = findByRole(wrapper, 'dialog');
      expect(modalElementHidden.isVisible()).toBe(false);
    });

    it('displays the started date with expected value', async () => {
      const dateElement = findByTestId(wrapper, 'fr-run-history-summary-modal-date');
      expect(dateElement.text()).toBe(dayjs('10/15/2023').format('MM/DD/YYYY h:mm A'));
    });

    it('displays the report status with expected value', async () => {
      const statusElement = findByTestId(wrapper, 'fr-complete-report-badge');
      expect(statusElement.text()).toBe('Complete');
    });
  });
});
