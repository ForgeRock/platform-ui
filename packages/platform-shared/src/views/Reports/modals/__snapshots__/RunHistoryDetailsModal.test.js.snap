// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Run History Details modal component @component emits "modal-toggle" when the "showModal" computed property is updated: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (2)",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_V_MODEL) v-model usage on component has changed in Vue 3. Component that expects to work with v-model should now use the "modelValue" prop and emit the "update:modelValue" event. You can update the usage and then opt-in to Vue 3 behavior on a per-component basis with \`compatConfig: { COMPONENT_V_MODEL: false }\`.
  Details: https://v3-migration.vuejs.org/breaking-changes/v-model.html",
    "
",
    " at <BModal",
    "modelValue=true",
    "onUpdate:modelValue=fn",
    "data-testid="run-history-details-modal"",
    " ...",
    ">",
  ],
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) (2)",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation OPTIONS_BEFORE_DESTROY) \`beforeDestroy\` has been renamed to \`beforeUnmount\`.",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_SCOPED_SLOTS) vm.$scopedSlots has been removed. Use vm.$slots instead.
  Details: https://v3-migration.vuejs.org/breaking-changes/slots-unification.html",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (3)",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_FUNCTIONAL) Functional component <BButtonClose> should be defined as a plain function in Vue 3. The "functional" option has been removed. NOTE: Before migrating to use plain functions for functional components, first make sure that all async components usage have been migrated and its compat behavior has been disabled.
  Details: https://v3-migration.vuejs.org/breaking-changes/functional-components.html",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_FUNCTIONAL) (2)",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) vm.$listeners has been removed. In Vue 3, parent v-on listeners are included in vm.$attrs and it is no longer necessary to separately use v-on="$listeners" if you are already using v-bind="$attrs". (Note: the Vue 3 behavior only applies if this compat config is disabled)
  Details: https://v3-migration.vuejs.org/breaking-changes/listeners-removed.html",
    "
",
    " at <BButtonClose",
    "onClick=fn<bound onClose>",
    "ref="close-button"",
    "content="&times;"",
    " ...",
    ">",
    "
",
    " at <Transition",
    "onBeforeEnter=fn<bound onBeforeEnter>",
    "onEnter=fn<bound onEnter>",
    "onAfterEnter=fn<bound onAfterEnter>",
    " ...",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (2)",
    "
",
    " at <BContainer",
    "class="p-0 text-dark"",
    ">",
    "
",
    " at <Transition",
    "onBeforeEnter=fn<bound onBeforeEnter>",
    "onEnter=fn<bound onEnter>",
    "onAfterEnter=fn<bound onAfterEnter>",
    " ...",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (3)",
    "
",
    " at <BListGroup",
    "horizontal=""",
    ">",
    "
",
    " at <BCol",
    "md="12"",
    ">",
    "
",
    " at <BRow",
    "no-gutters=""",
    ">",
    "
",
    " at <BContainer",
    "class="p-0 text-dark"",
    ">",
    "
",
    " at <Transition",
    "onBeforeEnter=fn<bound onBeforeEnter>",
    "onEnter=fn<bound onEnter>",
    "onAfterEnter=fn<bound onAfterEnter>",
    " ...",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (4)",
    "
",
    " at <BListGroupItem",
    "class="border-0 pt-0 pl-0 pr-5"",
    ">",
    "
",
    " at <BListGroup",
    "horizontal=""",
    ">",
    "
",
    " at <BCol",
    "md="12"",
    ">",
    "
",
    " at <BRow",
    "no-gutters=""",
    ">",
    "
",
    " at <BContainer",
    "class="p-0 text-dark"",
    ">",
    "
",
    " at <Transition",
    "onBeforeEnter=fn<bound onBeforeEnter>",
    "onEnter=fn<bound onEnter>",
    "onAfterEnter=fn<bound onAfterEnter>",
    " ...",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (4)",
    "
",
    " at <RunReportBadges",
    "report-status="complete"",
    ">",
    "
",
    " at <BListGroupItem",
    "class="border-0 pt-0"",
    ">",
    "
",
    " at <BListGroup",
    "horizontal=""",
    ">",
    "
",
    " at <BCol",
    "md="12"",
    ">",
    "
",
    " at <BRow",
    "no-gutters=""",
    ">",
    "
",
    " at <BContainer",
    "class="p-0 text-dark"",
    ">",
    "
",
    " at <Transition",
    "onBeforeEnter=fn<bound onBeforeEnter>",
    "onEnter=fn<bound onEnter>",
    "onAfterEnter=fn<bound onAfterEnter>",
    " ...",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_FUNCTIONAL) (3)",
    "
",
    " at <RunReportBadges",
    "report-status="complete"",
    ">",
    "
",
    " at <BListGroupItem",
    "class="border-0 pt-0"",
    ">",
    "
",
    " at <BListGroup",
    "horizontal=""",
    ">",
    "
",
    " at <BCol",
    "md="12"",
    ">",
    "
",
    " at <BRow",
    "no-gutters=""",
    ">",
    "
",
    " at <BContainer",
    "class="p-0 text-dark"",
    ">",
    "
",
    " at <Transition",
    "onBeforeEnter=fn<bound onBeforeEnter>",
    "onEnter=fn<bound onEnter>",
    "onAfterEnter=fn<bound onAfterEnter>",
    " ...",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (5)",
    "
",
    " at <BBadge",
    "key=1",
    "class="w-100"",
    "data-testid="fr-complete-report-badge"",
    " ...",
    ">",
    "
",
    " at <RunReportBadges",
    "report-status="complete"",
    ">",
    "
",
    " at <BListGroupItem",
    "class="border-0 pt-0"",
    ">",
    "
",
    " at <BListGroup",
    "horizontal=""",
    ">",
    "
",
    " at <BCol",
    "md="12"",
    ">",
    "
",
    " at <BRow",
    "no-gutters=""",
    ">",
    "
",
    " at <BContainer",
    "class="p-0 text-dark"",
    ">",
    "
",
    " at <Transition",
    "onBeforeEnter=fn<bound onBeforeEnter>",
    "onEnter=fn<bound onEnter>",
    "onAfterEnter=fn<bound onAfterEnter>",
    " ...",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (6)",
    "
",
    " at <BButton",
    "textContent="Close"",
    "onClick=fn<bound onOk>",
    "ref="ok-button"",
    " ...",
    ">",
    "
",
    " at <Transition",
    "onBeforeEnter=fn<bound onBeforeEnter>",
    "onEnter=fn<bound onEnter>",
    "onAfterEnter=fn<bound onAfterEnter>",
    " ...",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "appear" with v-bind value \`false\` will render appear="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <Transition",
    "onBeforeEnter=fn<bound onBeforeEnter>",
    "onEnter=fn<bound onEnter>",
    "onAfterEnter=fn<bound onAfterEnter>",
    " ...",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "persisted" with v-bind value \`false\` will render persisted="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <Transition",
    "onBeforeEnter=fn<bound onBeforeEnter>",
    "onEnter=fn<bound onEnter>",
    "onAfterEnter=fn<bound onAfterEnter>",
    " ...",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (7)",
    "
",
    " at <BVTransition",
    "noFade=false",
    ">",
    "
",
    " at <BModal",
    "data-testid="run-history-details-modal"",
    "ok-only=""",
    "ok-variant="outline-primary"",
    " ...",
    ">",
    "
",
    " at <RunHistoryDetailsModal",
    "isTesting=true",
    "parameters=",
    {
      "accountStatus": "accountStatus",
      "applications": [
        "applications",
      ],
      "campaign_name": "campaign_name",
      "campaign_status": "campaign_status",
      "endDate": "endDate",
      "journeyName": [
        "journeyName",
      ],
      "oauth2_applications": [
        "oauth2_applications",
      ],
      "org_names": [
        "org_names",
      ],
      "realm": "realm",
      "roles": [
        "roles",
      ],
      "startDate": "startDate",
      "status": "status",
      "treeName": [
        "treeName",
      ],
      "treeResult": "treeResult",
      "user_names": [
        "user_names",
      ],
    },
    "show=true",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_EVENT_EMITTER) vm.$on/$once/$off() have been removed. Use an external event emitter library instead.
  Details: https://v3-migration.vuejs.org/breaking-changes/events-api.html",
  ],
]
`;
